File: sdk/java/src/main/java/com/hjtpx/captcha/exception/ApiException.java
package com.hjtpx.captcha.exception;

public class ApiException extends CaptchaException {
    public ApiException(String message, String code) {
        super(message, code, isRetryableCode(code));
    }

    public ApiException(String message, String code, Throwable cause) {
        super(message, code, isRetryableCode(code), cause);
    }

    private static boolean isRetryableCode(String code) {
        if (code == null) return false;
        try {
            int statusCode = Integer.parseInt(code);
            return statusCode == 429 || (statusCode >= 500 && statusCode < 600);
        } catch (NumberFormatException e) {
            return false;
        }
    }
}

File: sdk/php/src/Exception/ApiException.php
<?php

namespace HJTPX\Captcha\Exception;

class ApiException extends CaptchaException
{
    protected $code;

    public function __construct(string $message, int $code = 0, ?\Throwable $previous = null)
    {
        $this->code = $code;
        parent::__construct($message, $code, $previous);
    }
}
